packages/@react-spectrum/s2/src/Breadcrumbs.tsx

```diff
@@ -71,43 +91,123 @@ const wrapper = style<BreadcrumbsStyleProps>({
   }
 }, getAllowedOverrides());
 
+const InternalBreadcrumbsContext = createContext<BreadcrumbsProps<any>>({});
+
 function Breadcrumbs<T extends object>(props: BreadcrumbsProps<T>, ref: DOMRef<HTMLOListElement>) {
   [props, ref] = useSpectrumContextProps(props, ref, BreadcrumbsContext);
+  let domRef = useDOMRef(ref);
   let {
     UNSAFE_className = '',
     UNSAFE_style,
     styles,
     size = 'M',
     children,
+    isDisabled,
     ...otherProps
   } = props;
-  let domRef = useDOMRef(ref);
+
   return (
-    <RACBreadcrumbs
-      {...otherProps}
-      ref={domRef}
-      style={UNSAFE_style}
-      className={UNSAFE_className + wrapper({
-        size
-      }, styles)}>
-      <BreadcrumbsContext.Provider value={props}>
-        {children}
-      </BreadcrumbsContext.Provider>
-    </RACBreadcrumbs>
+    <Provider
+      values={[
+        [InternalBreadcrumbsContext, {size, isDisabled}]
+      ]}>
+      <CollapsingCollection containerRef={domRef} onAction={props.onAction}>
+        <RACBreadcrumbs
+          {...otherProps}
+          isDisabled={isDisabled}
+          ref={domRef}
+          style={UNSAFE_style}
+          className={UNSAFE_className + wrapper({
+            size
+          }, styles)}>
+          {children}
+        </RACBreadcrumbs>
+      </CollapsingCollection>
+    </Provider>
   );
 }
 
+let BreadcrumbMenu = (props: {items: Array<Node<any>>, onAction: Pick<BreadcrumbsProps<unknown>, 'onAction'>}) => {
```

Comment:
```suggestion
let BreadcrumbMenu = (props: {items: Array<Node<any>>, onAction: BreadcrumbsProps<unknown>['onAction']}) => {
```

This should fix the `onAction` types mismatch

Reply:
ah thank you, i forgot about this syntax